superset-frontend/src/SqlLab/components/SqlEditor.jsx

```diff
@@ -222,6 +222,15 @@ class SqlEditor extends React.PureComponent {
     });
   }
 
+  static getDerivedStateFromProps(props, state) {
+    if (props.queryEditor.sql !== state.sql) {
+      return {
+        sql: props.queryEditor.sql,
+      };
+    }
+    return null;
+  }
```

Comment:
needing to use this function seems a bit concerning... i'm referencing here: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops:
> If you want to “reset” some state when a prop changes, consider either making a component fully controlled or fully uncontrolled with a key instead.

I wonder if we can just use the sql from the `queryEditor` for everything here instead of using the `state.sql` string?

Reply:
i think this is a good idea: I removed all the usage for `state.sql` and always use `props.queryEditor.sql`.